### ISSUE TITLE
Add access to the text events

### ISSUE THREAD

**Issue by bigspider** [LedgerHQ/ragger#180]
Sometimes it is desirable to test explicitly that a certain text string did appear on screen.

While screenshot comparison does implicitly test the screen text content, it would be nice in some cases to explicitly check conditions on the actual text that was shown on-screen. In fact, this would prevent the test from passing if an incorrect screenshot is inadvertently used in the test.

It could be as simple as an API call to get the list of all the text events.

**Comment by xchapron-ledger** [comment]
There is already the `get_current_screen_content` and `compare_screen_with_text` `BackendInterface` method that can be used for that.

**Comment by xchapron-ledger** [comment]
However, this will work only on Speculos, so tests using this won't be compatible with on device tests.

**Comment by bigspider** [comment]
I'm not very familiar with ragger, so I might be wrong - but I don't think it works well for my use case. See for example @sgliner-ledger's [ragger_bitcoin.py](https://github.com/LedgerHQ/app-bitcoin-new/blob/develop/ragger_bitcoin/ragger_bitcoin.py): it is using `navigate_until_text` and `navigate_until_text_and_compare`; what I would want is that the backend keeps track of all the text events, so that at the end of a UX test that used ragger, I have access to the list of text events. The text event could be one of the screen that `navigate_until_text_and_compare` moved past before stopping.

**Comment by xchapron-ledger** [comment]
Indeed, you currently can't retrieve the text events when you have navigated automatically through screens.
We could add something for that, but I'm not sure of why that matter? Mostly because when using `navigate_until_text_and_compare` the screen content is checked to be pixel perfect with the reference screenshot, so I don't see what should be retrieved in the screens.

**Comment by xchapron-ledger** [comment]
Well --golden_run shouldn't be ran with the eyes closed! IMO the changes can only be ignored if the diff was due to SDK changes.

Anyway, if you want this feature, indeed that's something to be added.
But:
- it obviously won't be compatible with on-device test
- the result will be different per device screen size, hence some checks discrepancy.
